utils.go

```diff
@@ -0,0 +1,58 @@
+package kinesis
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"text/template"
+
+	"github.com/gliderlabs/logspout/router"
+)
+
+type missingEnvVarError struct {
+	envVar   string
+	tmplName string
+}
+
+func (e *missingEnvVarError) Error() string {
+	return fmt.Sprintf("The %s template is missing. Please set the %s env variable\n", e.tmplName, e.envVar)
+}
+
+func compileTmpl(envVar string, tmplName string) (*template.Template, error) {
+	tmplString := os.Getenv(envVar)
+	if tmplString == "" {
+		return nil, &missingEnvVarError{envVar: envVar, tmplName: tmplName}
+	}
+
+	tmpl, err := template.New("").Parse(tmplString)
+	if err != nil {
+		return nil, err
+	}
+
+	return tmpl, nil
+}
+
+func executeTmpl(tmpl *template.Template, m *router.Message) (string, error) {
+	var res bytes.Buffer
+	err := tmpl.Execute(&res, m)
+	if err != nil {
+		return "", err
+	}
+
+	return res.String(), nil
+}
+
+func logErr(err error) {
+	if err != nil {
+		log.Println("kinesis: ", err.Error())
+	}
+}
+
+func logErrs(err []error) {
+	if err != nil {
+		for _, e := range err {
+			log.Println("kinesis: ", e.Error())
```

Comment:
I'd have this just call logErr from above, so that if the format changes you don't have to change both places